sql/src/main/java/org/apache/druid/sql/calcite/rel/Grouping.java

```diff
@@ -70,12 +71,25 @@ private Grouping(
       @Nullable final DimFilter havingFilter,
       final RowSignature outputRowSignature
   )
+  {
+    this(dimensions, subtotals, aggregations, havingFilter, outputRowSignature, false);
+  }
+
+  private Grouping(
+      final List<DimensionExpression> dimensions,
+      final Subtotals subtotals,
+      final List<Aggregation> aggregations,
+      @Nullable final DimFilter havingFilter,
+      final RowSignature outputRowSignature,
+      final boolean droppedDimensionsWhileApplyingProject
+  )
   {
     this.dimensions = ImmutableList.copyOf(dimensions);
     this.subtotals = Preconditions.checkNotNull(subtotals, "subtotals");
     this.aggregations = ImmutableList.copyOf(aggregations);
     this.havingFilter = havingFilter;
     this.outputRowSignature = Preconditions.checkNotNull(outputRowSignature, "outputRowSignature");
+    this.droppedDimensionsWhileApplyingProject = droppedDimensionsWhileApplyingProject;
```

Comment:
can we rename it to something else e.g. hadGroupingDimensions or groupDimensionsDropped or something? basically, we need not add `whileApplyingProject` here